providers/providers.go

```diff
@@ -116,6 +113,9 @@ func newProviderDataFromConfig(providerConfig options.Provider) (*ProviderData,
 			errs = append(errs, fmt.Errorf("could not parse %s URL: %v", name, err))
 		}
 	}
+	// handle LoginURLParameters
+	errs = p.compileLoginParams(errs, providerConfig.LoginURLParameters)
```

Comment:
Rather than passing errs in, I would, IMO it creates a nicer separation of concerns and means that a bug in compileLoginParams means we don't lose any errors
```suggestion
	errs = append(errs, p.compileLoginParams(providerConfig.LoginURLParameters)...)
```

Reply:
Fair, I think I was influenced by a pattern I'd seen somewhere in an earlier revision of this PR, will change.